### ISSUE TITLE
Closure compiler warnings

### ISSUE THREAD

**Issue by qw3n** [Websilk/Selector#3]
I threw you code into [closure compiler](https://closure-compiler.appspot.com) to see how it compressed it. And it comes up with a number of warnings. If you do the advanced compression it gives a couple extra warnings as well. Also the advanced compression comes back with a size of 2.76KB gzipped, but you would need to correct the warnings and test to see that it hasn't removed something. Anyways here is a list of warnings from the normal compression.

JSC_SUSPICIOUS_SEMICOLON: If this if/for/while really shouldn't have a body, use {} at line 71 character 69
                    for(var i = el.length; i--; elems.unshift(el[i]));
                                                                     ^
JSC_SUSPICIOUS_NAN: Comparison against NaN is always false. Did you mean isNaN()? at line 608 character 16
            if (n != NaN) { obj = n; } else {
                ^
JSC_UNREACHABLE_CODE: unreachable code at line 636 character 16
                return 0;
                ^
JSC_SUSPICIOUS_NAN: Comparison against NaN is always false. Did you mean isNaN()? at line 714 character 20
                if (padtop == NaN) { padtop = 0; }
                    ^
JSC_SUSPICIOUS_NAN: Comparison against NaN is always false. Did you mean isNaN()? at line 715 character 20
                if (padbot == NaN) { padbot = 0; }
                    ^
JSC_SUSPICIOUS_NAN: Comparison against NaN is always false. Did you mean isNaN()? at line 735 character 20
                if (padright == NaN) { padright = 0; }
                    ^
JSC_SUSPICIOUS_NAN: Comparison against NaN is always false. Did you mean isNaN()? at line 736 character 20
                if (padleft == NaN) { padleft = 0; }
                    ^
JSC_SUSPICIOUS_SEMICOLON: If this if/for/while really shouldn't have a body, use {} at line 751 character 60
            if (q.some(function (a) { return a == sel[0] }));
                                                            ^
JSC_USELESS_CODE: Suspicious code. The result of the 'eq' operator is not being used. at line 786 character 46
                    if (q.some(function (s) { s == n })) {elems.push(n);}
                                              ^
JSC_USELESS_CODE: Suspicious code. The result of the 'eq' operator is not being used. at line 1053 character 42
                if (q.some(function (s) { s == e.previousSibling })) {
                                          ^
JSC_USELESS_CODE: Suspicious code. This code lacks side-effects. Is there a bug? at line 1054 character 20
                    if (e.previousSibling) { elems.push[e.previousSibling]; }...
                    ^
JSC_USELESS_CODE: Suspicious code. The result of the 'getelem' operator is not being used. at line 1054 character 45
                    if (e.previousSibling) { elems.push[e.previousSibling]; }...
                                             ^
JSC_USELESS_CODE: Suspicious code. The result of the 'getelem' operator is not being used. at line 1054 character 85
...e.previousSibling) { elems.push[e.previousSibling]; } else { elems.push[e]; }
                                                                ^
JSC_USELESS_CODE: Suspicious code. This code lacks side-effects. Is there a bug? at line 1060 character 16
                if (e.previousSibling) { elems.push[e.previousSibling]; } els...
                ^
JSC_USELESS_CODE: Suspicious code. The result of the 'getelem' operator is not being used. at line 1060 character 41
                if (e.previousSibling) { elems.push[e.previousSibling]; } els...
                                         ^
JSC_USELESS_CODE: Suspicious code. The result of the 'getelem' operator is not being used. at line 1060 character 81
...e.previousSibling) { elems.push[e.previousSibling]; } else { elems.push[e]; }
                                                                ^
JSC_SUSPICIOUS_NAN: Comparison against NaN is always false. Did you mean isNaN()? at line 1512 character 16
            if (n != NaN) { obj = n; } else {
                ^
JSC_UNREACHABLE_CODE: unreachable code at line 1540 character 16
                return 0;
                ^

**Comment by markentingh** [comment]
Thank you for the insight! I will set up closure & fix these warnings, and hopefully we'll be able to reach 2.76KB gzipped :)

**Comment by markentingh** [comment]
I've updated selector.js in the repo, which can now compile using Google's closure compiler. The isssue I'm having now is that closure doesn't want to include a few of my prototype functions, such as:

`prototype.on, prototype.each, prototype.hasClass`

Also, even after closure missed some prototypes in the compiled code, the gzip compressed filesize doesn't compare to the gulp uglify compiler (at least for me). 

I get 4.99KB with closure and 4.65KB with uglify.

When changing the closure flag 'compilationLevel' to ADVANCED, the filesize is much smaller, but then all of my prototype functions are renamed. I fear there is no hope for closure.

**Comment by qw3n** [comment]
I was afraid that might be the case with the advanced compression. But I would look at the warnings I posted. For instance `NaN==NaN//returns false` so `padleft == NaN` is always going to be false.

**Comment by markentingh** [comment]
Yes, since the last push, I did fix all the bugs that closure insisted I should fix, and I added some JSDoc comments as well to help closure better understand the code structure. At this point, closure is just being stubborn, renaming prototype functions, even after I added special JSDoc comments to prevent closure from renaming them.
